File: src/01-easy/power-of-thor-episode-1/solution.py
lx, ly, tx, ty = [int(i) for i in input().split()]
while True:
    input()
    d = ""
    if ty > ly: d += "N"; ty -= 1
    elif ty < ly: d += "S"; ty += 1
    if tx > lx: d += "W"; tx -= 1
    elif tx < lx: d += "E"; tx += 1
    print(d)

File: src/01-easy/power-of-thor-episode-1/solution.pl
$| = 1;
chomp(my $line = <STDIN>);
my ($lx, $ly, $tx, $ty) = split / /, $line;
while (1) {
    <STDIN>;
    my $dir = "";
    if ($ty > $ly) { $dir .= "N"; $ty--; }
    elsif ($ty < $ly) { $dir .= "S"; $ty++; }
    if ($tx > $lx) { $dir .= "W"; $tx--; }
    elsif ($tx < $lx) { $dir .= "E"; $tx++; }
    print "$dir\n";
}
